### PR TITLE
Update output.md

### PR DESCRIPTION
**Pull request by xp44mm**
_describe your changes..._

- [ ] Read and sign the [CLA][1]. PRs that haven't signed it won't be accepted.
- [ ] Make sure your PR complies with the [writer's guide][2].
- [ ] Review the diff carefully as sometimes this can reveal issues.
- [ ] Do not abandon your Pull Request: [Stale Pull Requests][3].
- __Remove these instructions from your PR as they are for your eyes only.__

[1]: https://cla.js.foundation/webpack/webpack.js.org
[2]: https://webpack.js.org/contribute/writers-guide/
[3]: https://webpack.js.org/contribute/#pull-requests

### CODE REVIEW

**Comment by chenxsan** [comment]
Can you share the reason to update it?

File: src/content/configuration/output.md [hunk 1]
```diff
@@ -744,7 +744,7 @@ These options assign the return value of the entry point (e.g. whatever the entr
 If `output.library` is not assigned a non-empty string, the default behavior is that all properties returned by the entry point will be assigned to the object as defined for the particular `output.libraryTarget`, via the following code fragment:
 
 ```javascript
-(function(e, a) { for(var i in a) { e[i] = a[i]; } }(output.libraryTarget, _entry_return_));
+(function(e, a) { Object.assign(e, a); } }(output.libraryTarget, _entry_return_));
 ```
```
Comment (xp44mm) [inline src/content/configuration/output.md:748]: `Object.assign` had been a more common use, and It's self-evident

File: src/content/configuration/output.md [hunk 2]
```diff
@@ -903,10 +903,10 @@ The output will be:
...
     define([], factory);
-  else {
-    var a = factory();
-    for(var i in a) (typeof exports === 'object' ? exports : root)[i] = a[i];
-  }
+  else if(typeof exports === 'object')
+    Object.assign(exports, factory());
+  else
+    Object.assign(root, factory());
 })(typeof self !== 'undefined' ? self : this, function() {
```
Comment (xp44mm) [inline src/content/configuration/output.md:910]: Consistent with the structure of the code of named `MyLibrary`. and `Object.assign` is simpler.

File: src/content/guides/getting-started.md [hunk 3]
```diff
@@ -281,9 +281,8 @@ __package.json__
     "version": "1.0.0",
     "description": "",
     "scripts": {
--      "test": "echo \"Error: no test specified\" && exit 1"
-+      "test": "echo \"Error: no test specified\" && exit 1",
-+      "build": "webpack"
++      "build": "webpack",
+       "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (xp44mm) [inline src/content/guides/getting-started.md:285]: The two lines are the same except for the tail comma.

Comment (chenxsan) [reply src/content/guides/getting-started.md:285]: Please revert this change. If you're going to change this one, you need to update all the following guides listing `package.json`.

File: src/content/configuration/output.md (hunk 1 above)
Comment (chenxsan) [reply src/content/configuration/output.md:748]: I'm afraid we can't update it here, those code fragments are generated by webpack as per https://github.com/webpack/webpack/blob/master/lib/library/AssignLibraryPlugin.js#L146, if you think it's a good idea to update it, please create a PR over there.

File: src/content/configuration/output.md (hunk 2 above)
Comment (chenxsan) [reply src/content/configuration/output.md:910]: See the reason above.

**Comment by chenxsan** [comment]
Closing this for the listed answers above.

**Pull request closed without merging**